kafka-connect-s3/src/main/java/io/confluent/connect/s3/S3SinkConnectorConfig.java

```diff
@@ -660,6 +670,31 @@ public static ConfigDef newConfigDef() {
           Width.SHORT,
           "Enable Path Style Access to S3"
       );
+      configDef.define(
+          ELASTIC_BUFFER_ENABLE,
+          Type.BOOLEAN,
+          ELASTIC_BUFFER_ENABLE_DEFAULT,
+          Importance.LOW,
+          "Specifies whether or not to allocate Elastic buffer for staging s3-part to save memory.",
```

Comment:
We should also document about the potential performance impact (or high CPU usage) in the config documentation.

Reply:
I don't know what description is good for you. In my experience, the impact of performance and CPU usage is very slight. Would you have any suggesion?

Reply:
In fixed buffer, we would be filling the buffer once. copied X bytes into buffer. 
but for elastic, it would be X + X/2 + X/4 ... .  (for every part upload)
until denominator (2^y), y won't be that high generally, so its reasonable. I would say its a medium level impact. 


          "Specifies whether or not to allocate Elastic buffer for staging s3-part to save memory. Note that this may cause decreased performance or increased CPU usage" 
@pbadani @hsp8712  sounds good?

Reply:
[readding deleted comment] 
fixed buffer -> X bytes filled for a single part. 
elastic buffer -> X + X/2 + X/4 ... 
till denominator 2^y. I agree y won't be that high generally. But the impact is medium level imo. 

Suggesting config doc - 
`          "Specifies whether or not to allocate Elastic buffer for staging s3-part to save memory. Note that this may cause decreased performance or increased CPU usage" `
sounds good @hsp8712 @pbadani ?

Reply:
Ok, LGTM. I will modify it soon.